astroid/builder.py

```diff
@@ -261,8 +262,8 @@ def delayed_assattr(self, node):
             pass
 
 
-def build_namespace_package_module(name: str, path: list[str]) -> nodes.Module:
-    return nodes.Module(name, path=path, package=True)
+def build_namespace_package_module(name: str, path: Sequence[str]) -> nodes.Module:
+    return nodes.Module(name, path=list(path), package=True)
```

Comment:
Should we type as sequence in Module constructor too ? I'm not sure of the performance implications of this cast.

Reply:
Shouldn't be too bad I think.

I'll add a TODO. I think this cascades to a lot of other things and even potentially to `pylint`. Because `mypy` can't be turned on right now it's quite difficult to do the updating everywhere were needed.

Reply:
```suggestion
    # TODO: Typing: Remove the cast to list and just update typing to accept Sequence
    return nodes.Module(name, path=list(path), package=True)
```

Reply:
Right this could be huge, let's wait for proper type check before going crazy.